codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientInterface.java

```diff
@@ -114,22 +132,38 @@ private BuilderModelBag operationSignatureAndJavaDoc(OperationModel opModel) {
                                                      .addParameter(requestType, opModel.getInput().getVariableName())
                                                      .addJavadoc(opModel.getAsyncDocumentation(model, opModel));
 
+        streamingMethod(methodBuilder, opModel, responsePojoType);
+
+        return new BuilderModelBag(methodBuilder, opModel, false);
+    }
+
+    private BuilderModelBag simpleMethod(OperationModel opModel) {
+        ClassName responsePojoType = ClassName.get(modelPackage, opModel.getReturnType().getReturnType());
```

Comment:
It feels like there's a fair amount of duplication between this and the `operationMethod` method - but maybe optimising it will add further complexity.